src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -50,38 +52,56 @@ public void showTotalBasket() {
 
     public boolean findProduct(String productName) {
         for (Product product : products) {
-            if ((product != null) && (productName.equals(product.getProductName()))) {
-                System.out.println("У вас уже есть такой товар");
+            if ((product != null) && (product.getProductName().equalsIgnoreCase(productName))) {
+                System.out.println("Товар " + product.getProductName() + " найден!");
                 return true;
             }
         }
         System.out.println("Такого товара нет");
         return false;
     }
 
-    public void showSpecialsQuantity(){
-        System.out.println("Количество специальных товаров - "+countSpecial()+" позиций");
+    public void showSpecialsQuantity() {
+        System.out.println("Количество специальных товаров - " + countSpecial() + " позиций");
     }
 
 
     public void cleanBasket() {
-        Arrays.fill(products, null);
+        products.clear();
         System.out.println("Корзина пуста");
     }
 
-    public void addToBasket(Product product){
+    public void addToBasket(Product product) {
         if (product != null) {
-            if (size < products.length) {
-                products[size] = product;
-                size++;
-                System.out.println("Продукт "+product.getProductName()+" успешно добавлен в корзину");
-            } else {
-                System.out.println("Невозможно добавить продукт "+product.getProductName()+". Корзина переполнена.");
-            }
+            products.add(product);
+            System.out.println("Товар " + product.getProductName() + " успешно добавлен в корзину");
         } else {
             System.out.println("Попытка добавить несуществующий продукт в корзину!");
         }
     }
 
+    public ArrayList<Product> removeFromBasket(String productToRemove) {
+        Iterator<Product> iterator = products.iterator();
+        ArrayList<Product> removedProduct = new ArrayList<>();
```

Comment:
тип ссылки должен быть максимально общим, чтобы использовать преимущества полиморфизма
```suggestion
        List<Product> removedProduct = new ArrayList<>();
```